### PR TITLE
improve C API documentation

### PR DESCRIPTION
**Pull request by mgovers**
Very small improvements to C API documentation regarding columnar data.

Relates to #548

### CODE REVIEW

File: power_grid_model_c/power_grid_model_c/include/power_grid_model_c/dataset.h [hunk 1]
```diff
@@ -140,7 +140,7 @@ PGM_API void PGM_destroy_dataset_const(PGM_ConstDataset* dataset);
  *     If the component is not uniform, indptr must point to an array of size (batch_size + 1).
  *         The values in the array must be not decreasing.
  *         And we must have indptr[0] = 0, indptr[batch_size] = total_elements.
- * @param data A void pointer to the row based buffer data or a nullptr for columnar data.
+ * @param data A void pointer to the row based buffer data or NULL for columnar data.
```
Comment (mgovers) [inline power_grid_model_c/power_grid_model_c/include/power_grid_model_c/dataset.h:143]: C (until C23) has no `nullptr`

File: power_grid_model_c/power_grid_model_c/include/power_grid_model_c/buffer.h [hunk 2]
```diff
@@ -20,15 +20,15 @@ extern "C" {
...
+ * with size and alignment obtained from PGM_meta_component_size() and PGM_meta_component_alignment().
  * The buffer created by this function should be freed by PGM_destroy_buffer().
  *
  * It is recommended to call PGM_buffer_set_nan() after you create an input or update buffer.
  * In this way all the attributes will be set to NaN.
- * And if there is a new optional attribute added in the future.
- * You have garantee that your code is still compatible
+ * Additionally, if there is a new optional attribute added in the future,
+ * compatibility with your code is guaranteed,
  * because that optional attribute will be set to NaN and the default value will be used.
```
Comment (Jerry-Jinfeng-Guo) [inline power_grid_model_c/power_grid_model_c/include/power_grid_model_c/buffer.h:32]:
```suggestion
 * compatibility with your code is guaranteed: that optional attribute will be set to NaN 
 * and the default value will be used.
```

**Comment by mgovers** [comment]
> While going through this PR, two questions were raised:
> 
>     * Since we don't have a create buffer function for columnar buffers (since probably there is no value in it due to the nature of columnar data), we should document this somewhere as there is an "inconsistency" on how row and columnar data are managed at the C API level. Or at least those are my thought, it may lead to confusion from the user perspective.

I've also been thinking about that, but I don't know where to put it. Do you have a good proposal?

>     * Should we (at some point) make something like `is_columnar` available through the C API? It may not be needed, but given that we offer functions to check if something is batch and its size, it would probably go along with those (and may even provide value in the python side).

I think that makes sense, but maybe out of scope of `v1.10.0`. @TonyXiang8787 do you have any thoughts on this?

**Review by Jerry-Jinfeng-Guo — APPROVED** [review]
Ongoing discussions are deemed out of scope of this PR. Please create a dedicated thread to discuss them further.

**Comment by figueroa1395** [comment]
> > While going through this PR, two questions were raised:
> > 
> > * Since we don't have a create buffer function for columnar buffers (since probably there is no value in it due to the nature of columnar data), we should document this somewhere as there is an "inconsistency" on how row and columnar data are managed at the C API level. Or at least those are my thought, it may lead to confusion from the user perspective.
> 
> I've also been thinking about that, but I don't know where to put it. Do you have a good proposal?

Not sure yet, it probably makes sense to directly comment something about it under `PGM_create_buffer`, but it should probably be standalone somewhere else.

**Comment by mgovers** [comment]
continuing thread in https://github.com/PowerGridModel/power-grid-model/issues/548#issuecomment-2428421384

**Pull request merged**